### PR TITLE
Move docs inline from helpdb and add a couple doctests

### PR DESCRIPTION
**Pull request by xorJane**
I moved some docstrings inline, added a doctest for `reverseind`, and converted the examples listed in the docstring for `signif` to doctests.

Thank you and happy weekend!

Ref #22521

### CODE REVIEW

File: base/abstractarray.jl [hunk 1]
```diff
@@ -842,6 +842,12 @@ Represents the array `y` as an array having the same indices type as `x`.
 """
 of_indices(x, y) = similar(dims->y, oftype(indices(x), indices(y)))
 
+
+"""
+    full(F)
+
+Reconstruct the matrix `A` from the factorization `F=factorize(A)`.
+"""
```
Comment (Sacha0) [inline base/abstractarray.jl]: (Tangent: This docstring for `full` only captures one of the various meanings of `full`. Perhaps we should flesh out this docstring? Or not worry about it in anticipation of complete excision of full in this release cycle?)

File: base/strings/types.jl [hunk 2]
```diff
@@ -127,6 +127,23 @@ reverse(s::RevString) = s.string
...
+Given an index `i` in `reverse(v)`, return the corresponding index in `v` so that
+`v[reverseind(v,i)] == reverse(v)[i]`. (This can be nontrivial in the case where `v` is a
+Unicode string.)
+
+```jldoctest
+julia> r = reverse("Julia")
+"ailuJ"
+
+julia> for i in 1:length(r)
+           print(r[reverseind(s, i)])
```
Comment (Sacha0) [inline base/strings/types.jl]: Mysterious `s`? :)

File: base/floatfuncs.jl [hunk 3]
```diff
@@ -131,6 +131,20 @@ function _signif_og(x, digits, base)
     return og, e
 end
 
+"""
+    signif(x, digits, [base])
+
+Rounds (in the sense of [`round`](@ref)) `x` so that there are `digits` significant digits, under a
+base `base` representation, default 10.
+
+```jldoctest
```
Comment (tkelman) [inline base/floatfuncs.jl]:
```
# Examples
```jldoctest
```

File: base/strings/types.jl [hunk 4]
```diff
@@ -127,6 +127,23 @@ reverse(s::RevString) = s.string
...
 ## reverse an index i so that reverse(s)[i] == s[reverseind(s,i)]
 
+"""
+    reverseind(v, i)
+
+Given an index `i` in `reverse(v)`, return the corresponding index in `v` so that
+`v[reverseind(v,i)] == reverse(v)[i]`. (This can be nontrivial in the case where `v` is a
+Unicode string.)
+
+```jldoctest
```
Comment (tkelman) [inline base/strings/types.jl]:
```
# Examples
```jldoctest
```

File: base/initdefs.jl [hunk 5]
```diff
@@ -18,6 +18,12 @@ An array of the command line arguments passed to Julia, as strings.
 """
 const ARGS = String[]
 
+"""
+    exit([code])
+
+Quit (or control-D at the prompt). The default exit code is zero, indicating that the
+processes completed successfully.
```
Comment (tkelman) [inline base/initdefs.jl]: does processes being plural here sound redundant / a little off to anyone else?

Comment (Sacha0) [inline base/initdefs.jl]: Sounded a little off on first read, yes. Technically more correct than a singular equivalent though I suppose.

On that note, the "control-D at the prompt" snippet strikes me as out of place as well. Perhaps could be listed as a synonym instead? Also could use a ref to the `quit()` docstring.

File: base/strings/types.jl (hunk 2 above)
Comment (xorJane) [inline base/strings/types.jl]: Fixed!

File: base/strings/types.jl (hunk 4 above)
Comment (xorJane) [inline base/strings/types.jl]: I added `# Examples` to a few other places in the files I touched.

**Review by Sacha0 — APPROVED** [review]
lgtm (the tangential comments aside)! :)

File: base/abstractarray.jl (hunk 1 above)
Comment (ararslan) [inline base/abstractarray.jl]: If it were me I'd just leave it as-is, since that's a larger task than simply relocating the docstring.

**Comment by Sacha0** [comment]
(Absent objections, requests for time, or someone beating me to it, I plan to merge this tomorrow morning. Best!)

File: base/initdefs.jl (hunk 5 above)
Comment (tkelman) [inline base/initdefs.jl]: we should maybe note these as future todo cleanups somewhere

Comment (Sacha0) [inline base/initdefs.jl]: Captured in an intro issue: #22765. Best!

**Pull request merged**